backend.py

```diff
@@ -1,13 +1,146 @@
-from flask import Flask
-from config import connect_db
-db = connect_db()
-from pony.orm import db_session
+from decimal import Decimal
+
+import numpy as np
+import pandas as pd
+from flask import Flask, jsonify
+from flask_cors import CORS
+
+from helpers import txs
+from inputs import address_inputs, method_input
 
 app = Flask(__name__)
+CORS(app)
 
-@app.route("/")
+@app.route("/", methods=['POST'])
 def hello_yrs_backend():
-    with db_session:
-      result = db.select('NOW()')
-      return f"<p>Hello, yrs-backend!</p><p>it's {result[0]}</p>"
+    def process_sale():
+        nonlocal taxable_events
```

Comment:
what's the purpose of `nonlocal` here ?

Reply:
I've very significantly restructured the yrs backend, many of these will have been changed by now. 
nonlocal taxable_events allows process_sale() to modify the taxable_events variable even though it resides outside of the function